### ISSUE TITLE
Review of combinatorial evidence branch

### ISSUE THREAD

**Issue by jbmunro** [evidenceontology/evidenceontology#189]
Stemming from NTR for combinatorial evidence #188, we began reviewing the combinatorial evidence branch.  Notes and suggestions follow.

Input and suggestions welcomed.

James

**Notes from our meeting:**
* We would be willing to create a term to the effect of “combinatorial evidence from computational and experimental evidence” but not a more specific child that replaces “computational evidence” with a more specific method such as the software Matrisome Annotator.  Creating terms with such specifics could potentially lead to a proliferation of such terms.  Additionally, ECO:0000245 is a crossproduct term and we do not assert children for these terms.  (More generalized manual and automatic assertion terms may have children inferred.)
* ECO:0000053 needs to, at the very least, have the definition improved – pull from comments as they make it more clear as it included experimental and computational evidence.
* ECO:0000053 as defined, assumes auto assertion (“...evaluated by an algorithm”).  (Were manual and auto assertion children added after this term was created?  Yes.  These children were automatically generated and without reviewing the parent definition, the fact that a redundant term was created was probably missed.)
* Are ECO:0000053 and its automatic assertion child redundant?  Yes.
* Are the following terms redundant?  
	* ECO:0000053 computational combinatorial evidence
	* ECO:0000246 computational combinatorial evidence used in automatic assertion
	* ECO:0000213 combinatorial evidence used in automatic assertion
As defined currently, it strikes me that ECO:0000053 and ECO:0000246 are redundant.  I don’t believe that ECO:0000213 is redundant with either term as it fails to capture “computational” evidence.

* Look at other children of ‘combinatorial evidence’ which are very clear and consider the label for ECO:0000053 - i.e. do we want to generate more specific terms?
	* combinatorial evidence from curator knowledge and experimental evidence
	* combinatorial evidence from author knowledge and experimental evidence
* Note that ECO:0000245 computational combinatorial evidence used in manual assertion has GO term RCA attached.  It is imperative that the parent reflects the GO meaning.  At the time we suggested that GO:RCA is possibly supposed to be a child of ECO:0000244 ‘combinatorial evidence used in manual assertion’, however see below.
* Note that ECO:0000245 doesn’t make sense as the parent definition (ECO:0000053) states that it is evaluated by an algorithm.
 

**From Michelle:**
“It looks like RCA means when the evidence types are integrated computationally and then the results are manually reviewed.”

Rachael needs to annotate to GO:RCA, which is attached to the current term ECO:0000245 ‘computational combinatorial evidence used in manual assertion’.

As noted above, we could conceivably transfer GO:RCA to ECO:0000244 ‘combinatorial evidence used in manual assertion’; however, in my opinion, it fails to capture the definition of RCA as provided above by Michelle.

**ECO:0000053 computational combinatorial evidence = TAIR:TED term**
**Def:** “A type of combinatorial analysis where data are combined and evaluated by an algorithm.”
**Note:** comments refer to both computational and experimental results.
**Note:** definition implies “automatic assertion” rather than a combination of experimental and computational evidence.
**Note:** child ECO:0000245 ‘computational combinatorial evidence used in manual assertion’ is tagged with GO:RCA which does not belong here – see above def. for RCA.

**Modify term and definition for ECO:0000053 or obsolete the term and creating a new one**

**Old name:** 'computational combinatorial evidence'
**New name:** ‘combinatorial evidence from evidence that is computationally integrated’

**Modified def:** “A type of combinatorial analysis where computational and/or experimental data are computationally combined.”

**Children:**
ECO:0000245 ‘computational combinatorial evidence used in manual assertion’ would become ‘combinatorial evidence from evidence that is computationally integrated used in manual assertion’

While it doesn’t explicitly state that the results are “manually reviewed” as defined in GO:RCA, *maybe* it is implied as it is in all of our “* manual assertion” terms.  (My initial thinking is that manually (or computationally) integrated data could be potentially manually or automatically asserted.)

ECO:0000246 ‘computational combinatorial evidence used in automatic assertion’ would become ‘combinatorial evidence from evidence that is computationally integrated used in automatic assertion’

Removing “...and evaluated by an algorithm.” from the parent definition makes this term not redundant.

Then, we need to discuss generating new terms like:
‘combinatorial evidence from evidence that is manually integrated’
‘combinatorial evidence from experimental and computational evidence’
‘combinatorial evidence from experimental evidence only’
‘combinatorial evidence from computational evidence only’

**Notes from Michelle**
In looking at the “combinatorial” terms more, we found what we feel is a problem.

Here are the terms:
-ECO:0000212 ‘combinatorial evidence’ with definition: “A type of evidence that arises out of the integration of more than one line of evidence.”
---ECO:0000053 ‘computational combinatorial evidence’ with definition: “A type of combinatorial analysis where data are combined and evaluated by an algorithm.” The comment indicates that evidence could be experimental or computational.
-----ECO:0000245 ‘computational combinatorial evidence used in manual assertion’ with definition: “A type of computational combinatorial analysis that is used in a manual assertion.” This term has no comment. This term is mapped to GO RCA evcode.
-----ECO:0000246 ‘computational combinatorial evidence used in automatic assertion’ with def: “A type of computational combinatorial analysis that is used in an automatic assertion.” This term has no comment.
---ECO:0000213 ‘combinatorial evidence used in automatic assertion’ with def: “A type of combinatorial analysis that is used in an automatic assertion.” Again, comment states can be experimental or computational evidence.
---ECO:0000244 ‘‘combinatorial evidence used in manual assertion’ with def: “A type of combinatorial analysis that is used in a manual assertion.” Again, comment states can be experimental or computational evidence.

When I look at the def and comment for 53, appears to me that this term is indicating that the assertion step is automated – it certainly states that the evaluation is automated. Therefore, I actually feel that 53, 246, and 213 are all basically equivalent. And that 245 should be equivalent to 244. Also that 245 should not be a child of 53 – as it’s basically saying a manual assertion is a child of an automatic assertion.

245 is currently mapped to RCA.

When I look at the GO’s documentation for RCA, it seems that there are two steps it is referring to:
1. the step of combining the multiple evidences
2. the step of making assertions from the results.

Currently our terms do not reflect these two differences. Things are munged together and thus creating some of the issues outlined above.

Question: do we want the terms to reflect automatic/manual options for both of these steps?

**Comment by ivanerill** [comment]
>As defined currently, it strikes me that ECO:0000053 and ECO:0000246 are redundant. I don’t believe that ECO:0000213 is redundant with either term as it fails to capture “computational” evidence.

Quick question. ECO:0000053 and ECO:0000246 may be logically redundant, but woudn't you still want a properly derived cross-product, to normalize the usage of the term? And, thinking through it, you could conceive that an author overrides or accepts the conclusion of a comp. combinational method, hence leading to a ECO:0000245. So, if ECO:0000245 is valid, then ECO:0000246 must be too.

>Removing “...and evaluated by an algorithm.” from the parent definition makes this term not redundant.

Makes sense to me

> “It looks like RCA means when the evidence types are integrated computationally and then the results are manually reviewed.”

It's a bit hard to parse, but yes, they explicitly say that non-curator/author reviewed computational pipelines should get IEA.

If this is the case, though, ECO:0000246 cannot map to RCA. Only ECO:0000245 should. This brings up the issue of text-mining as an example in the definition of ECO:0000053

> And that 245 should be equivalent to ECO:0000244.

ECO:0000245 should be more specific than ECO:0000244.  But it is not. The comments are copy-pasted from ECO:0000053 into ECO:0000245, not ECO:0000244. That raises the point of whether there is "combinatorial" evidence in the absence of a computer. If so, then cases should be identified and added to theECO:0000 244 comments. The current examples seem to apply exclusively to what is should be covered by ECO:0000053 & ECO:0000245.

> Question: do we want the terms to reflect automatic/manual options for both of these steps?

I believe you want to reflect whether the data juggling/wrangling is done by a computer or not in the term, although I don't know how relevant that is. How the assertion is done is covered by the cross-product and I believe it should be made explicit here too for consistency.

**Comment by rctauber** [comment]
We discussed this issue on an ECO group call today and have decided on a proposal for this problem.

Here is a general outline:

1. Create a new 'automatically integrated combinatorial evidence' class.
2. Change `ECO:0000053` to 'automatically integrated combinatorial evidence used in automatic assertion' and merge `ECO:0000246` into this.
3. Change `ECO:0000245` to 'automatically integrated combinatorial evidence used in manual assertion' and keep the RCA mapping on this one.
4. Create manually-integrated combinatorial evidence classes, matching the above.
5. Create a new 'combinatorial computational evidence' class, with automatically- and manually-integrated children. The automatically-integrated, used in manual assertion child maps to RCA.
6. Create a new 'combinatorial experimental evidence' class, same as above.
7. Create a new 'combinatorial computational and experimental evidence' class, same as above (all automatically-integrated, used in manual assertion classes will map to RCA).
8. Create a new 'computational evidence' class to use in logical definitions for the combinatorial classes. Review other classes at a later time to make logical connections.

I've included a file for review here, it's a bit easier to follow with all the definitions :smile:
Run the reasoner to see all inferred relationships that will be asserted in the ECO release!
[combinatorial_evidence.owl.zip](https://github.com/evidenceontology/evidenceontology/files/2650170/combinatorial_evidence.owl.zip)

Please let me know if there are any questions, or if I can clarify anything.

**Comment by jbmunro** [comment]
Looking really good!  No issues that I can see.

Question - 'automatically integrated combinatorial evidence used in manual assertion' will become ECO:0000245 GO:RCA - right?

**Comment by rctauber** [comment]
Yup @jbmunro - that is correct! It will be a child of 'automatically integrated combinatorial evidence'.

**Comment by mgiglio99** [comment]
@rctauber - quick confirmation - above in #5
"Create a new 'combinatorial computational evidence' class, with automatically- and manually-integrated children. The automatically-integrated, used in manual assertion child maps to RCA."
When you say "child maps to RCA" - you mean child actually is ECO 245 - correct?

@ivanerill 
These combinatorial terms have their roots in the GO RCA code.
As described above, our reading of the GO RCA documentation is that there are two steps being captured with that code - the step of combining the different evidence types, and the step of making an assertion based on the results. 
BAsed on our reading of the definition for ECO 53 which currently is:
"A type of combinatorial analysis where data are combined and evaluated by an algorithm." 
we feel this term is representing automated steps for both the combination and the results evaluation. So we will be refining the def of this term and placing it properly in the new structure that reflects these two steps. When that happens, we find that 246 and 53 are the same.

We do not currently have examples for the manually integrated terms - at least that I know of.
A fear I have is that RCA might actually be being used by GO for both manual and automatic integration. I'm hoping some go people will chime in on this.

@rachhuntley 
This new structure will provide you with the first new term you requested in issue #188 , but not the second one. Please take a look at this and see if it makes sense to you.

Hoping for input from @pgaudet and @cmungall

**Comment by rctauber** [comment]
@mgiglio99 
> The automatically-integrated, used in manual assertion child maps to RCA."
When you say "child maps to RCA" - you mean child actually is ECO 245 - correct?
Yes, that is correct.

**Comment by jbmunro** [comment]
Wrapping this up - changes to the combinatorial evidence branch have been pushed.

**Comment by rachhuntley** [comment]
Hi James,
Sorry, I think I misunderstood. 

I thought there would be a new term, which @mgiglio99 said would be the first term I requested in #188 (ECO:new combinatorial evidence from computational and experimental evidence used in manual assertion) that I assumed would be this one from the outline above:

7. Create a new 'combinatorial computational and experimental evidence' class, same as above (all automatically-integrated, used in manual assertion classes will map to RCA).

But now I'm wondering what automatically integrated means vs. manually integrated? In my use case, does it mean that the Matrisome analysis tool I use (described in #188) has automatically integrated data?

**Comment by jbmunro** [comment]
Hi @rachhuntley,

Pinging @rctauber and @mgiglio99 

No apology necessary.  Our review of that branch changed things quite a bit so your confusion is justified.  Automatically integrated would infer that the data / evidence were combined automatically - by a computer program for example.  I do believe that your use of the Matrisome analysis tool fits.

James

**Comment by mgiglio99** [comment]
@rctauber can you point Rachael to the id of the new term?

**Comment by jbmunro** [comment]
@rctauber  and @mgiglio99 

I thought that I had pointed Rachael to the id for the term that maps to GO:RCA above - 'automatically integrated combinatorial evidence used in manual assertion' - ECO:0000245.

James

**Comment by rachhuntley** [comment]
Hi James,
It is the ID of the new child term of ECO:245 that I want (if I'm understanding this correctly, I can't see the actual terms that were created):
ECO:0000245 'automatically integrated combinatorial evidence used in manual assertion' 
- ECO:new 'automatically-integrated combinatorial computational and experimental evidence used in manual assertion'

Copying in @RLovering to follow up on this, as this is my last working day.

**Comment by rctauber** [comment]
I think the term you are looking for is ECO:0007665, 'automatically
integrated combinatorial computational and experimental evidence', is that
correct? RCA is still mapped to the term that James pointed to.

**Comment by rachhuntley** [comment]
That sounds correct. Is this a child of ECO:245 so will map up to RCA anyway? Also, does it have the 'used in manual assertion' at the end?
Thanks.

**Comment by jbmunro** [comment]
@rctauber, @mgiglio99, @RLovering, @rachhuntley

Ah - thanks for pointing Rachael to the more specific term.  I pointed her to ECO:0000245 because of the original request to map to GO:RCA and that is the term with the exact synonym.  

Becky - I'm not familiar with the greater mapping file so please confirm... 

ECO:7665 is a child of 'automatically integrated combinatorial evidence':ECO:0007673 and 'automatically integrated combinatorial evidence used in manual assertion':ECO:0000245 so it should map to ECO:0000245 - right?

Rachael, ECO:7665 has a child 'automatically integrated combinatorial computational and experimental evidence used in manual assertion' - ECO:ECO:0007666.

James

**Comment by rachhuntley** [comment]
Great, thanks James. It looks like ECO:7666 is the one I need, as long as it maps to ECO:245.

**Comment by rachhuntley** [comment]
Thanks everyone!

**Comment by RLovering** [comment]
Hi 
I am concerned about the parents for ECO:0007666. I think ECO:0007667 should be a sibling not a parent to this term.

ECO:0007666 automatically integrated combinatorial computational and experimental evidence used in manual assertion

ECO:0007667 automatically integrated combinatorial computational and experimental evidence used in automatic assertion

Thanks

Ruth

**Comment by rctauber** [comment]
Hi @RLovering - we saw this in the end-of-March release and it should be fixed in the April 1, 2019 release. Please take a look at this and let me know if you have any other questions!

**Comment by RLovering** [comment]
OK brilliant, thanks for spotting it and for confirming so quickly

Ruth
